scripts/eosio_build_ubuntu.sh

```diff
@@ -1,430 +1,269 @@
+		if [ "${OS_MAJ}" -lt 18 ]; then
+			printf "You must be running Linux Mint 18.x or higher to install EOSIO.\\n"
+			printf "Exiting now.\\n"
+			exit 1
+		fi
+	;;
+	"Ubuntu")
+		if [ "${OS_MAJ}" -lt 16 ]; then
+			printf "You must be running Ubuntu 16.04.x or higher to install EOSIO.\\n"
+			printf "Exiting now.\\n"
+			exit 1
+		fi
+		# UBUNTU 18 doesn't have MONGODB 3.6.3
+		if [ $OS_MAJ -gt 16 ]; then
+			export MONGODB_VERSION=4.1.1
+		fi
+		# We have to re-set this with the new version
+		export MONGODB_ROOT=${OPT_LOCATION}/mongodb-${MONGODB_VERSION}
+	;;
+	"Debian")
+		if [ $OS_MAJ -lt 10 ]; then
+			printf "You must be running Debian 10 to install EOSIO, and resolve missing dependencies from unstable (sid).\n"
+			printf "Exiting now.\n"
+			exit 1
 	fi
+	;;
+esac
 
-	case "${OS_NAME}" in
-		"Linux Mint")
-		   if [ "${OS_MAJ}" -lt 18 ]; then
-			   printf "\\tYou must be running Linux Mint 18.x or higher to install EOSIO.\\n"
-			   printf "\\tExiting now.\\n"
-			   exit 1
-		   fi
-		;;
-		"Ubuntu")
-			if [ "${OS_MAJ}" -lt 16 ]; then
-				printf "\\tYou must be running Ubuntu 16.04.x or higher to install EOSIO.\\n"
-				printf "\\tExiting now.\\n"
+if [ "${DISK_AVAIL%.*}" -lt "${DISK_MIN}" ]; then
+	printf "You must have at least %sGB of available storage to install EOSIO.\\n" "${DISK_MIN}"
+	printf "Exiting now.\\n"
+	exit 1
+fi
+
+# llvm-4.0 is installed into /usr/lib/llvm-4.0
+# clang is necessary for building on ubuntu
+DEP_ARRAY=(
+	git llvm-4.0 clang-4.0 libclang-4.0-dev make automake libbz2-dev libssl-dev doxygen graphviz \
+	libgmp3-dev autotools-dev build-essential libicu-dev python2.7 python2.7-dev python3 python3-dev \
+	autoconf libtool curl zlib1g-dev sudo ruby
+)
+COUNT=1
+DISPLAY=""
+DEP=""
+
+if [[ "${ENABLE_CODE_COVERAGE}" == true ]]; then
+	DEP_ARRAY+=(lcov)
+fi
+
+if [ $1 == 0 ]; then read -p "Do you wish to update repositories with apt-get update? (y/n) " answer; fi
+case ${answer} in
+	1 | [Yy]* )
+		if ! apt-get update; then
+			printf " - APT update failed.\\n"
+			exit 1;
+		else
+			printf " - APT update complete.\\n"
+		fi
+	;;
+	[Nn]* ) echo "Proceeding without update!";;
+	* ) echo "Please type 'y' for yes or 'n' for no.";;
+esac
+
+printf "\\nChecking for installed dependencies...\\n"
+for (( i=0; i<${#DEP_ARRAY[@]}; i++ )); do
+	pkg=$( dpkg -s "${DEP_ARRAY[$i]}" 2>/dev/null | grep Status | tr -s ' ' | cut -d\  -f4 )
+	if [ -z "$pkg" ]; then
+		DEP=$DEP" ${DEP_ARRAY[$i]} "
+		DISPLAY="${DISPLAY}${COUNT}. ${DEP_ARRAY[$i]}\\n"
+		printf " - Package %s${bldred} NOT${txtrst} found!\\n" "${DEP_ARRAY[$i]}"
+		(( COUNT++ ))
+	else
+		printf " - Package %s found.\\n" "${DEP_ARRAY[$i]}"
+		continue
+	fi
+done
+if [ "${COUNT}" -gt 1 ]; then
+	printf "\\nThe following dependencies are required to install EOSIO:\\n"
+	printf "${DISPLAY}\\n\\n" 
+	if [ $1 == 0 ]; then read -p "Do you wish to install these packages? (y/n) " answer; fi
+	case ${answer} in
+		1 | [Yy]* )
+			if ! apt-get -y install ${DEP}; then
```

Comment:
The removal of sudo here means that as a normal user (i.e. not root) I can't just `./eosio_build.sh` if I'm missing a dependency. I guarantee this indirectly encourages users to run the script as root (bad).

If we,
* Don't want sudo in scripts (good practice IMO but I'm not staunchly against it)
* Don't want users running and building as root (never good to encourage this, very dangerous)

Then I think we might need to change this to instructions instead, like print out something:
> You need to install some additional dependencies to build eosio, run sudo apt-get install blah blah

I'm also fine with adding back the sudo if we prefer that approach. But as-is this is troublesome.

Reply:
Sudo added back